### PR TITLE
PT-2836

### PR DESCRIPTION
**Pull request by allasm**
Better response when trying to save pedigree as a user with not enough rights (e.g. non-logged in user)

### CODE REVIEW

**Comment by allasm** [comment]
@veronikaslc @a-gvo please review

File: components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/PhenotipsFamilyTools.java [hunk 1]
```diff
@@ -195,7 +192,26 @@ public boolean currentUserCanDeleteFamily(String familyId, boolean deleteAllMemb
     }
 
     @Override
-    public boolean currentUserHasAccessRight(Family family, Right right)
+    public boolean familyExists(String familyId)
+    {
+        Family family = this.familyRepository.getFamilyById(familyId);
+        if (family == null) {
```
Comment (veronikaslc) [inline components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/PhenotipsFamilyTools.java]: `return family != null'` ?

File: components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/PhenotipsFamilyTools.java [hunk 2]
```diff
@@ -195,7 +192,26 @@ public boolean currentUserCanDeleteFamily(String familyId, boolean deleteAllMemb
...
+    }
+
+    @Override
+    public boolean currentUserHasAccessRight(String familyId, Right right)
+    {
+        Family family = this.familyRepository.getFamilyById(familyId);
+        if (family == null) {
+            return false;
+        }
+        return this.currentUserHasAccessRight(family, right);
```
Comment (allasm) [inline components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/PhenotipsFamilyTools.java]: +1, I removed a bunch of extra checks but missed/added this

File: components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/PhenotipsFamilyTools.java (hunk 1 above)
Comment (allasm) [inline components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/PhenotipsFamilyTools.java]: +1

**Comment by allasm** [comment]
Addressed comments and force-pushed new version

File: components/family-studies/rest/src/main/java/org/phenotips/studies/family/rest/internal/DefaultFamiliesResourceImpl.java [hunk 3]
```diff
@@ -182,7 +182,9 @@ public Response listFamilies(Integer start, Integer number, Boolean fullJSON)
                         } else {
                             familyJSON = getFamilySummaryJSON(record, this.uriInfo);
                         }
-                        familyList.put(familyJSON);
+                        if (familyJSON != null) {
```
Comment (sdumitriu) [inline components/family-studies/rest/src/main/java/org/phenotips/studies/family/rest/internal/DefaultFamiliesResourceImpl.java]: Now this breaks checkstyle, too many branches in this method. Can you look at it?

Comment (sdumitriu) [inline components/family-studies/rest/src/main/java/org/phenotips/studies/family/rest/internal/DefaultFamiliesResourceImpl.java]: @allasm

**Pull request merged**